File: web/src/api/orderfood/recommendation.js
import { orderFoodRequest } from './request'

const encodePath = (value) => encodeURIComponent(value)

/**
 * 分页查询可发现候选菜品。
 */
export const getDiscoverableDishList = (params) =>
  orderFoodRequest({
    path: '/discoverable-dishes',
    method: 'get',
    params
  })

/**
 * 获取可发现候选菜品详情。
 */
export const getDiscoverableDishDetail = (dishId, params) =>
  orderFoodRequest({
    path: `/discoverable-dishes/${encodePath(dishId)}`,
    method: 'get',
    params
  })

/**
 * 分页查询推荐精选。
 */
export const getRecommendationList = (params) =>
  orderFoodRequest({
    path: '/recommendations',
    method: 'get',
    params
  })

/**
 * 获取推荐精选详情。
 */
export const getRecommendationDetail = (recommendationId) =>
  orderFoodRequest({
    path: `/recommendations/${encodePath(recommendationId)}`,
    method: 'get'
  })

/**
 * 创建推荐草稿。
 */
export const createRecommendation = (data) =>
  orderFoodRequest({
    path: '/recommendations',
    method: 'post',
    data,
    mutation: true
  })

/**
 * 编辑推荐排序和展示说明。
 */
export const updateRecommendation = (recommendationId, data) =>
  orderFoodRequest({
    path: `/recommendations/${encodePath(recommendationId)}`,
    method: 'put',
    data,
    mutation: true
  })

/**
 * 删除推荐草稿。
 */
export const deleteRecommendation = (recommendationId, data) =>
  orderFoodRequest({
    path: `/recommendations/${encodePath(recommendationId)}`,
    method: 'delete',
    data,
    mutation: true
  })

/**
 * 发布或重新发布推荐。
 */
export const publishRecommendation = (recommendationId, data) =>
  orderFoodRequest({
    path: `/recommendations/${encodePath(recommendationId)}/publish`,
    method: 'post',
    data,
    mutation: true
  })

/**
 * 下线推荐。
 */
export const offlineRecommendation = (recommendationId, data) =>
  orderFoodRequest({
    path: `/recommendations/${encodePath(recommendationId)}/offline`,
    method: 'post',
    data,
    mutation: true
  })

/**
 * 批量调整推荐排序。
 */
export const updateRecommendationSortOrder = (data) =>
  orderFoodRequest({
    path: '/recommendations/sort-order',
    method: 'put',
    data,
    mutation: true
  })

File: server/router/dish/recommendation.go
package dish

import (
	"github.com/dyjh/order-food-mini-app/server/middleware"
	"github.com/gin-gonic/gin"
)

// RecommendationRouter 提供可发现候选菜品和推荐精选路由注册能力。
type RecommendationRouter struct{}

// InitRecommendationRouter 注册可发现候选菜品和推荐精选管理路由。
func (router *RecommendationRouter) InitRecommendationRouter(orderFoodGroup *gin.RouterGroup) {
	group := orderFoodGroup
	mutationGroup := orderFoodGroup.Group("").Use(middleware.OperationRecord())
	{
		group.GET("/discoverable-dishes", recommendationApi.ListDiscoverableDishes)
		group.GET("/discoverable-dishes/:dishId", recommendationApi.GetDiscoverableDish)

		group.GET("/recommendations", recommendationApi.ListRecommendations)
		group.GET("/recommendations/:recommendationId", recommendationApi.GetRecommendation)
		mutationGroup.POST("/recommendations", recommendationApi.CreateRecommendation)
		mutationGroup.PUT("/recommendations/sort-order", recommendationApi.UpdateRecommendationSortOrder)
		mutationGroup.PUT("/recommendations/:recommendationId", recommendationApi.UpdateRecommendation)
		mutationGroup.DELETE("/recommendations/:recommendationId", recommendationApi.DeleteRecommendation)
		mutationGroup.POST("/recommendations/:recommendationId/publish", recommendationApi.PublishRecommendation)
		mutationGroup.POST("/recommendations/:recommendationId/offline", recommendationApi.OfflineRecommendation)
	}
}
